tests: drivers: i2s: i2s_api: drivers.i2s.gpio_loopback fails

**Describe the bug**
The tests `drivers.i2s.gpio_loopback` fails.

Observed for:
- `nrf52840dk_nrf52840`
- `nrf5340dk_nrf5340_cpuapp`.

**To Reproduce**
Steps to reproduce the behavior:
1. have `nrf52840dk_nrf52840` connected
2. go to your zephyr dir
3. call `./scripts/twister -T tests/net/lib/lwm2m/content_plain_text -p nrf52840dk_nrf52840 --device-testing --device-serial /dev/ttyACM0 -v --inline-logs --fixture gpio_loopback`
4. See console output with error

**Expected behavior**
Valid console output

**Impact**
Not clear

**Logs and console output**
```
zephyr/zephyr/drivers/i2s/i2s_handlers.c: In function 'z_vrfy_i2s_configure':
zephyr/zephyr/drivers/i2s/i2s_handlers.c:36:48: error: 'struct k_mem_slab' has no member named 'block_size'
   36 |         if (config.block_size > config.mem_slab->block_size) {
      |  
```

**Environment (please complete the following information):**
 - OS: Ubuntu 20.04.1 LTS
 - Toolchain Zephyr SDK 0.16.0
 - Commit SHA or Version used: zephyr-v3.4.0-4796-gd44e96e486ae

**Additional context**
Used set of pins:
```
psels = <NRF_PSEL(I2S_SCK_M, 1, 5)>,
		<NRF_PSEL(I2S_LRCK_M, 1, 6)>,
		<NRF_PSEL(I2S_SDOUT, 1, 1)>,
		<NRF_PSEL(I2S_SDIN, 1, 2)>;
```

--- a/drivers/i2s/i2s_handlers.c
+++ b/drivers/i2s/i2s_handlers.c
@@ -33,7 +33,7 @@ static inline int z_vrfy_i2s_configure(const struct device *dev,
 	/* Ensure that the k_mem_slab's slabs are large enough for the
 	 * specified block size
 	 */
-	if (config.block_size > config.mem_slab->block_size) {
+	if (config.block_size > config.mem_slab->info.block_size) {
 		goto out;
 	}
 

